svm/src/transaction_processor.rs

```diff
@@ -259,11 +283,161 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
                                 compute_budget_process_transaction_time.as_us()
                             );
                             if let Err(err) = maybe_compute_budget {
+                                Self::update_loaded_accounts(
```

Comment:
if a transaction is not executed, it (before simd82) cannot be in a block and thus it should have no affect on state.

What are you trying to update here? To me it seems extremely dangerous.

Reply:
This is due to the same reason as mentioned in (https://github.com/anza-xyz/agave/pull/1468#discussion_r1618198482)
I think my assumption is not correct, let me remove the update_loaded_accounts call for all the failing transactions cases.

Reply:
addressed the new points here [be317eac6cf3417c3369e30f713bef9a98f88c51], please review.